src/service/history-impl.js

```diff
@@ -590,6 +636,12 @@ export class HistoryBindingVirtual_ {
   }
 
   /** @override */
+  replaceStateForTarget(target) {
+    let hash = target.indexOf('#') == 0 ? target : `#${target}`;
```

Comment:
We actually dropped this statement all together and now just assert that it starts with a `#`. Github is probably showing you a previous change.